pyomo/contrib/parmest/parmest.py

```diff
@@ -279,20 +299,37 @@ def __init__(
         assert isinstance(experiment_list, list)
         self.exp_list = experiment_list
 
-        # check that an experiment has experiment_outputs and unknown_parameters
-        model = self.exp_list[0].get_labeled_model()
-        try:
-            outputs = [k.name for k, v in model.experiment_outputs.items()]
-        except:
-            RuntimeError(
-                'Experiment list model does not have suffix ' + '"experiment_outputs".'
-            )
-        try:
-            params = [k.name for k, v in model.unknown_parameters.items()]
-        except:
-            RuntimeError(
-                'Experiment list model does not have suffix ' + '"unknown_parameters".'
-            )
+        # check that each model has "get_labeled_model()" function, and
+        # at the same time count the number of parameters for sampling
+        param_list = []
+        for i in range(len(self.exp_list)):
+
+            try:
+                model = self.exp_list[i].get_labeled_model()
+            except:
+                RuntimeError(
+                    "Experiment list model "
+                    + str(i)
+                    + ' is missing ".get_labeled_model()" function.'
+                )
+
+            # check that each model has experiment_outputs
+            # (although this is optional for design of experiments)
+            outputs = getattr(model, "experiment_outputs", None)
+            if outputs is None:
+                logger.warning(
+                    "Experiment outputs missing for experiment " + str(i) + "."
+                )
+
+            # check that each model has unknown parameters (not optional)
+            params = getattr(model, "unknown_parameters", None)
+            assert isinstance(params, Suffix)
```

Comment:
`assert`s like this are problematic because, if the assertion isn't true, the user won't get a helpful message. Can this be changed to something that raises a helpful error message?